### PR TITLE
add ability to apply ad hoc edits to vcs command line

### PR DESCRIPTION
**Pull request by chick**
the vcs command generated by testers has a number of flags by default and can be augmented by the --more-vcs-flags and --more-vcs-c-flags options.  This PR provides a mechanism for removing existing flags or applying other regex based edits to the command line.  It is a bandage fix, the more proper (and harder) fix would be to somehow expose the full construction of the vcs command to users.

### CODE REVIEW

File: src/main/scala/chisel3/iotesters/CommandEditor.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              show(s"""$messagePrefix no match for "$editPatternString" on command "$currentCommandLine" """)
+          }
+        case CommandEditor.Verbose() =>
+          verbose = true
+          show(s"""$messagePrefix applying edits to "$currentCommandLine" """)
+        case _ =>
+          show(s"""vcs-command-edit ignoring edit command "$line" """)
+      }
+    }
+    currentCommandLine
```
Comment (ucbjrl) [inline src/main/scala/chisel3/iotesters/CommandEditor.scala]: Should there be some kind of sanity/security check that we haven't generated a command line such as:
```bash
% cd mydir && vcs -quiet -timescale=1ns/1ps -debug_pp ; mail $USER -s "You've been pwned" && encrypt /home
```
We should at least warn that this mechanism is insecure and has the potential to be abused.

Comment (ucbjrl) [inline src/main/scala/chisel3/iotesters/CommandEditor.scala]: It should be noted that this was possible before the introduction of the `CommandEditor`. The ability to supply unchecked flags is sufficient.

**Comment by stevenmburns** [comment]
Works great once I figured out how to escape the substitution string.
Anything better than what I did below?

````
class GenericFuncTestVCS( tag : String, factory : () => FifoIfc)
    extends FlatSpec with Matchers {
  behavior of tag

  val optionsManager = new TesterOptionsManager {
    testerOptions =
      testerOptions.copy(
        backendName = "vcs",
        vcsCommandEdits = """s/\+vcs\+initreg\+random //"""
      )
  }

  it should "work" in {
    chisel3.iotesters.Driver.execute( factory, optionsManager) { c =>
      new FuncTester(c)
    } should be (true)
  }
}
````

**Comment by chick** [comment]
@stevenmburns I think you did the simplest way.  The regex's may have been overkill, but I wanted a mechanism that that would be pretty general despite being a hack

File: src/main/scala/chisel3/iotesters/CommandEditor.scala (hunk 1 above)
Comment (chick) [inline src/main/scala/chisel3/iotesters/CommandEditor.scala]: Where do you think the right place to put that message would be, in the command line help?
I don't quite understand the last comment about unchecked flags

**Pull request merged**